### PR TITLE
feature: add ssl.set_http_version API

### PR DESCRIPTION
**Pull request by vislee**
I hereby granted the copyright of the changes in this pull request
to the authors of this lua-resty-core project.

Note:  the unit test relying on the [test-nginx/pull/68](https://github.com/openresty/test-nginx/pull/68)

### CODE REVIEW

File: t/http2.t [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    lua_package_path "$TEST_NGINX_LUA_PACKAGE_PATH/?.lua;;";
+
+
+--- config
+    ssl_certificate      ../../cert/test.crt;
+    ssl_certificate_key  ../../cert/test.key;
+
+    ssl_certificate_by_lua_block {
+        local ssl = require "ngx.ssl";
+        ssl.set_http_version(2);
```
Comment (agentzh) [inline t/http2.t]: I think it's better to use the `ssl.use_http2(true)` and `ssl.use_http2(false)` here instead. We cannot really set other HTTP version numbers like 0.9, 1.0, and 1.1 here anyway.

File: t/http2.t [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+
+=== TEST 1: test set http2
+--- http_config
+    lua_package_path "$TEST_NGINX_LUA_PACKAGE_PATH/?.lua;;";
+
+
+--- config
+    ssl_certificate      ../../cert/test.crt;
+    ssl_certificate_key  ../../cert/test.key;
```
Comment (agentzh) [inline t/http2.t]: Better let the test scaffold automatically generate these under the hood. It's too much a burden for us test writers to write these in every HTTP/2 over TLS test cases ;)

**Comment by agentzh** [comment]
@vislee I wonder if the protocol selection result in the TLS handshake can be remembered by subsequent new connections doing TLS session resumption (session ID or session ticket). I doubt it though. Have you tested the TLS session resumption case?

**Comment by lziest** [comment]
I feel we need to really test session resumption, because in case of session resumption, the certificate callback will be skipped entirely. So I am a little skeptical about this PR right now.

**Comment by agentzh** [comment]
@lziest Agreed. We need test cases for different kinds of session resumptions, SSL session IDs and TLS session tickets.

**Comment by vislee** [comment]
@agentzh  @lziest  Thanks for review and comments. I will try to modify it. @agentzh  you can close this PR. Sorry!

**Comment by agentzh** [comment]
@vislee You can close it yourself, BTW :) Closing this.

**Pull request closed without merging**